firebase/rules/tests/idx.spec.ts

```diff
@@ -0,0 +1,129 @@
+    expect(anonymousUser).cannotWrite('/idx/user_labs/1', '');
+    expect(currentUser).cannotWrite('/idx/user_labs/1', '');
+    expect(systemUser).cannotWrite('/idx/user_labs/1', '');
+  });
+
+  it('All users can read into idx lab executions', () => {
+    expect(anonymousUser).canRead('/idx/lab_executions/1');
+    expect(currentUser).canRead('/idx/lab_executions/1');
+    expect(systemUser).canRead('/idx/lab_executions/1');
+  });
+
+  it('No users can write into idx lab executions', () => {
+    expect(anonymousUser).cannotWrite('/idx/lab_executions/1', '');
+    expect(currentUser).cannotWrite('/idx/lab_executions/1', '');
+    expect(systemUser).cannotWrite('/idx/lab_executions/1', '');
+  });
+
+  it('All users can read into idx visible lab executions', () => {
+    expect(anonymousUser).canRead('/idx/lab_visible_executions/1');
+    expect(currentUser).canRead('/idx/lab_visible_executions/1');
+    expect(systemUser).canRead('/idx/lab_visible_executions/1');
+  });
+
+  it('No users can write into idx visible lab executions', () => {
+    expect(anonymousUser).cannotWrite('/idx/lab_visible_executions/1', '');
+    expect(currentUser).cannotWrite('/idx/lab_visible_executions/1', '');
+    expect(systemUser).cannotWrite('/idx/lab_visible_executions/1', '');
+  });
+
+  it('Only system-user read into idx user executions', () => {
+    expect(anonymousUser).cannotRead('/idx/user_executions/1');
+    expect(currentUser).cannotRead('/idx/user_executions/1');
+    expect(systemUser).canRead('/idx/user_executions/1');
+  });
+
+  it('No users can write into idx user executions', () => {
+    expect(anonymousUser).cannotWrite('/idx/user_executions/1', '');
+    expect(currentUser).cannotWrite('/idx/user_executions/1', '');
+    expect(systemUser).cannotWrite('/idx/user_executions/1', '');
+  });
+
+  it('All users can read into idx visible user executions', () => {
+    expect(anonymousUser).canRead('/idx/user_visible_executions/1');
+    expect(currentUser).canRead('/idx/user_visible_executions/1');
+    expect(systemUser).canRead('/idx/user_visible_executions/1');
+  });
+
+  it('No users can write into idx visible user executions', () => {
+    expect(anonymousUser).cannotWrite('/idx/user_visible_executions/1', '');
+    expect(currentUser).cannotWrite('/idx/user_visible_executions/1', '');
+    expect(systemUser).cannotWrite('/idx/user_visible_executions/1', '');
+  });
+
+  it('All users can read into idx visible user labs', () => {
+    expect(anonymousUser).canRead('/idx/user_visible_labs/1');
+    expect(currentUser).canRead('/idx/user_visible_labs/1');
+    expect(systemUser).canRead('/idx/user_visible_labs/1');
+  });
+
+  it('No users can write into idx visible user labs', () => {
+    expect(anonymousUser).cannotWrite('/idx/user_visible_labs/1', '');
+    expect(currentUser).cannotWrite('/idx/user_visible_labs/1', '');
+    expect(systemUser).cannotWrite('/idx/user_visible_labs/1', '');
+  });
+});
```

Comment:
Thanks for the coverage here. I wonder if we should merge all the test sets a la "users can read but not write into XXX" and then have all the expectations in a single spec instead.